src/rhsm/profile.py

```diff
@@ -75,7 +75,14 @@ def _uniquify(module_list):
         ret = {}
         for module in module_list:
             key = (module["name"], module["stream"], module["version"], module["context"], module["arch"])
-            ret[key] = module
+            # Prefer duplicates that are Active.
+            # There are "enabled" duplicates from dnf that are marked as inactive.
+            if ret.get(key) is not None:
+                if ret[key].get("active", False) is False:
+                    ret[key] = module
+            else:
+                ret[key] = module
```

Comment:
The code is IMHO not fully correct. I prefer something like this:

```python
if key not in ret:
   ret[key] = module
else:
    if ret[key].get("active", False) is False and module.get("active", False) is True:
        ret[key] = module
```

It IMHO make sense to rewrite existing value only in the case, when module is active.